pkg/collection/entry/metadata.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package entry
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// Metadata provides mime type and filename to file entry.
+type Metadata struct {
+	MimeType string
```

Comment:
why do you need to export these fields? you use getters in the entry type, why not here?

Reply:
The entry type has getters because they need to implement an interface. Happy to put getters here too, even if no interface is implemented, if you think it's better?

Reply:
Actually, there are no getters for these fields. I would keep it simple, exported fields are the simplest. :)

Reply:
cool with exporting, but then we dont get promise of immutability